packages/xstate-react/test/useMachine.test.tsx

```diff
@@ -338,4 +339,53 @@ describe('useMachine hook', () => {
     // Just testing that it compiles
     render(<App />);
   });
+
+  it('should capture all actions', (done) => {
+    let count = 0;
+
+    const machine = createMachine({
+      initial: 'active',
+      states: {
+        active: {
+          on: {
+            EVENT: {
+              actions: () => {
+                count++;
+              }
+            }
+          }
+        }
+      }
+    });
+
+    const App = () => {
+      const [stateCount, setStateCount] = useState(0);
+      const [state, send] = useMachine(machine, {
+        execute: false
+      });
+
+      React.useEffect(() => {
+        send('EVENT');
+        send('EVENT');
+      }, []);
+
+      React.useEffect(() => {
+        setStateCount((c) => c + 1);
+      }, [state]);
+
+      return <div data-testid="count">{stateCount}</div>;
+    };
+
+    const { getByTestId } = render(<App />);
+
+    const countEl = getByTestId('count');
+
+    act(() => {
```

Comment:
`act` is only needed when you schedule some React updates and you want to have them flushed within act's callback, in here you just make assertions which is unrelated to React and thus this `act` doesn't do anything special and can be avoided